src/2d/spread2d_wrapper.cu

```diff
@@ -586,6 +586,7 @@ int CUSPREAD2D_SUBPROB_PROP(int nf1, int nf2, int M, CUFINUFFT_PLAN d_plan)
 	MapBintoSubProb_2d<<<(numbins[0]*numbins[1]+1024-1)/1024, 1024>>>(
 			d_subprob_to_bin,d_subprobstartpts,d_numsubprob,numbins[0]*numbins[1]);
 	assert(d_subprob_to_bin != NULL);
+        if (d_plan->subprob_to_bin != NULL) cudaFree(d_plan->subprob_to_bin);
```

Comment:
Shouldn't the assert above catch the NULL? I don't see why the if is needed.

Reply:
The if statement is checking the `d_plan->subprob_to_bin` pointer. The next line (not showing here), we assign `d_subprob_to_bin` to  `d_plan->subprob_to_bin`.